### PR TITLE
Ana2k/color picker

### PR DESCRIPTION
**Pull request by Ana2k**
This PR fixes the color pallette not appearing issue.

https://github.com/Igalia/wolvic/issues/422

Image for the same :
<table>
<tr>
  <td>
  </td>
</tr>
</table>

### CODE REVIEW

**Comment by Ana2k** [comment]
Tagging @felipeerias for code review.

File: app/src/main/res/layout/prompt_color.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            app:layout_constraintBottom_toTopOf="@id/buttonsLayout"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/title"
+            app:palette="@drawable/palette"
+            app:preferenceName="MyColorPicker" />
+
+
+
+        <LinearLayout
```
Comment (felipeerias) [inline app/src/main/res/layout/prompt_color.xml:42]: These buttons should follow the same style as in other similar dialogs, like `prompt_confirm.xml`, `prompt_file.xml`, etc.

File: app/src/common/shared/com/igalia/wolvic/browser/PromptDelegate.java [hunk 2]
```diff
@@ -533,4 +551,4 @@ public void onSessionChanged(@NonNull Session aOldSession, @NonNull Session aSes
         cleanSession(aOldSession);
         setUpSession(aSession);
     }
-}
+}
```
Comment (felipeerias) [inline app/src/common/shared/com/igalia/wolvic/browser/PromptDelegate.java:554]: This line change is not needed.

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/ColorPromptWidget.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            }
+            hide(REMOVE_WIDGET);
+        });
+
+        mOkButton.setOnClickListener(v -> {
+            if (mAudio != null) {
+                mAudio.playSound(AudioEngine.Sound.CLICK);
+            }
+            if (mPromptDelegate != null && mPromptDelegate instanceof ColorPromptDelegate) {
+                String colorCodeHex = "#" + colorPickerView.getColorEnvelope().getHexCode().substring(2, 8);
```
Comment (felipeerias) [inline app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/ColorPromptWidget.java:68]: Could you add a comment here explaining what this code does?

File: app/src/main/res/layout/prompt_color.xml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            android:layout_width="325dp"
+            android:layout_height="325dp"
+            android:layout_margin="10dp"
+            app:layout_constraintBottom_toTopOf="@id/buttonsLayout"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/title"
+            app:palette="@drawable/palette"
+            app:preferenceName="MyColorPicker" />
+
```
Comment (felipeerias) [inline app/src/main/res/layout/prompt_color.xml:39]: Unneeded extra lines.

**Review by felipeerias — CHANGES_REQUESTED** [review]
Thank you very much @Ana2k 

The PR looks good to me, there are just a few small issues left.

File: app/src/common/shared/com/igalia/wolvic/browser/PromptDelegate.java (hunk 2 above)
Comment (Ana2k) [reply app/src/common/shared/com/igalia/wolvic/browser/PromptDelegate.java:554]: For this line, I think the code added to the file above has shifted the last brace.
No changes from my end are fixing it.

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/ColorPromptWidget.java (hunk 3 above)
Comment (Ana2k) [reply app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/ColorPromptWidget.java:69]: Added the same.

File: app/src/main/res/layout/prompt_color.xml (hunk 4 above)
Comment (Ana2k) [reply app/src/main/res/layout/prompt_color.xml:39]: Removed.

File: app/src/main/res/layout/prompt_color.xml (hunk 1 above)
Comment (Ana2k) [reply app/src/main/res/layout/prompt_color.xml]: Updated.

File: app/src/main/res/layout/prompt_color.xml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:id="@+id/buttonsLayout"
+            android:layout_width="wrap_content"
```
Comment (felipeerias) [inline app/src/main/res/layout/prompt_color.xml:42]: One last thing: this line needs to be replaced with:

```
android:layout_width="match_parent"
```

This change aligns the buttons to the right of the dialog window, like in other similar prompts.

For example, this is how the confirmation prompt looks like now:

com igalia wolvic-20230413-154306

And this is how the color picker prompt will look with the suggested change:

com igalia wolvic-20230413-154236

**Review by felipeerias — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/prompt_color.xml (hunk 5 above)
Comment (Ana2k) [reply app/src/main/res/layout/prompt_color.xml:42]: Sure will do 👍
Thanks !😊

Comment (felipeerias) [reply app/src/main/res/layout/prompt_color.xml]: Thank you!

**Review by felipeerias — APPROVED** [review]
LGTM

**Pull request merged**